ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -1,88 +1,98 @@
-use std::{
-    collections::HashMap,
-    ops::Deref,
-    path::PathBuf,
-    str::FromStr,
-    sync::{Arc, RwLock},
-};
+use std::{path::PathBuf, str::FromStr, sync::Arc};
 
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 use axum::{
     body::boxed,
     http::{header, Request, Uri},
     response::{IntoResponse, Response},
     Json,
 };
 use hyper::Body;
+use kv::{Bucket, Config, Store};
 use serde::{Deserialize, Serialize};
 use tabby_common::{config::RepositoryConfig, SourceFile, Tag};
 use tower::ServiceExt;
 use tower_http::services::ServeDir;
-use tracing::{debug, error, warn};
 
-use crate::{
-    cron::{CronEvents, StartListener},
-    schema::repository::RepositoryService,
-};
+use crate::schema::repository::RepositoryService;
+
+fn repository_meta_db() -> PathBuf {
+    tabby_common::path::tabby_root().join("repositories.kv")
+}
 
+#[derive(Clone)]
 pub struct RepositoryCache {
-    repository_lookup: RwLock<HashMap<RepositoryKey, RepositoryMeta>>,
-    service: Arc<dyn RepositoryService>,
+    cache: Store,
 }
 
 impl std::fmt::Debug for RepositoryCache {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("RepositoryCache")
-            .field("repository_lookup", &self.repository_lookup)
-            .finish()
+        f.debug_struct("RepositoryCache").finish()
     }
 }
+type RepositoryBucket<'a> = Bucket<'a, String, kv::Json<RepositoryMeta>>;
 
 impl RepositoryCache {
-    pub async fn new_initialized(
-        service: Arc<dyn RepositoryService>,
-        events: &CronEvents,
-    ) -> Arc<RepositoryCache> {
-        let cache = RepositoryCache {
-            repository_lookup: Default::default(),
-            service,
-        };
-        if let Err(e) = cache.reload().await {
-            error!("Failed to load repositories: {e}");
-        };
-        let cache = Arc::new(cache);
-        cache.start_reload_listener(events);
-        cache
+    pub fn new() -> Result<Self> {
+        let config = Config::new(repository_meta_db());
+        let store = Store::new(config)?;
+        Ok(RepositoryCache { cache: store })
     }
 
-    async fn reload(&self) -> Result<()> {
-        let new_repositories = self
-            .service
-            .list_repositories(None, None, None, None)
-            .await?
-            .into_iter()
-            .map(|repository| RepositoryConfig::new_named(repository.name, repository.git_url))
-            .collect();
-        let mut repository_lookup = self.repository_lookup.write().unwrap();
-        debug!("Reloading repositoriy metadata...");
-        *repository_lookup = load_meta(new_repositories);
+    pub fn latest_version(&self) -> Result<String> {
+        let bucket = self.cache.bucket(Some("version"))?;
```

Comment:
```suggestion
        let bucket = self.cache.bucket(Some("meta"))?;
```

also replace raw string of `version`, `meta` into static const &str